components/src/core/DropdownToolbar/DropdownToolbar.tsx

```diff
@@ -0,0 +1,181 @@
+import React, { useState, useCallback, useRef } from 'react';
+import {
+    Typography,
+    Toolbar,
+    ListItemText,
+    Menu,
+    MenuItem,
+    ToolbarProps,
+    createStyles,
+    makeStyles,
+    useTheme,
+    Theme,
+} from '@material-ui/core';
+import { ArrowDropDown } from '@material-ui/icons';
+import clsx from 'clsx';
+
+const MENU_ITEM_HEIGHT = 48;
+
+export type ToolbarMenuItem = {
+    label: string;
+    onClick: () => void;
+};
+export type DropdownToolbarProps = ToolbarProps & {
+    title: string;
+    subtitleLabel: string | undefined;
+    menuItems?: ToolbarMenuItem[] | undefined;
+    navigationIcon?: JSX.Element;
+    customMenu?: JSX.Element;
+    classes?: DropdownToolbarClasses;
+};
+
+const useStyles = makeStyles((theme: Theme) =>
+    createStyles({
+        root: {
+            paddingLeft: theme.spacing(2),
+            paddingRight: theme.spacing(2),
+        },
+        navigationIconWrapper: {
+            marginRight: theme.spacing(4),
+            cursor: 'pointer',
+        },
+        title: {
```

Comment:
Looks like the title/subtitle is not doing ellipsing when the content is too long